source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown

```diff
@@ -0,0 +1,183 @@
+  twitter_handle: marpo60
+  github_handle: marpo60
+  image: /images/team/marcelo-dominguez.jpg
+  description: Software engineer - Curious by nature - Sports lover - Tech guy
+---
+
+## Introduction
+
+###What's the authentication agent forwarding for?
+
+Lets start showing what `man ssh` says about it:
+
+> `-A` Enables forwarding of the authentication agent
+> connection. This can also be specified on a per-host basis in a
+> configuration file.
+Agent forwarding should be enabled with caution.  Users with the
+ability to bypass file permissions on the remote host (for the agent's
+UNIX-domain socket)
+can access the local agent through the forwarded connection.  An
+attacker cannot obtain key material from
+the agent, however they can perform operations on the keys that
+enable them to authenticate using the identities loaded into the
+agent.
+
+It tries to solve the following problem:
+
+`Laptop -> Server A -> Server B`
+
+User wants to connect to Server A and from Server A to Server B
+forwarding all authentication requests to Laptop.
+
+<!--more-->
+
+### How do we use it?
+
+{% codeblock lang:sh %}
+  user@laptop$ ssh -A serverA
+  user@serverA$ ssh serverB
+  user@serverB$
+{% endcodeblock %}
+
+## So, When ssh -A is insecure?
+
+The man page has the following note:
+
+> Agent forwarding should be enabled with caution.
+
+So, we have to be carefull with this option, but what are the problems?
+
+With the intention of gaining more insight about the problem. We'll do
+an experiment, we'll set up an scenario that is far form ideal but it's
+simple enough to reproduce the problem.
+
+## Experiment
+
+To follow this experiment you'll need to have [vagrant](https://www.vagrantup.com/) installed
+in your computer. You don't need to have previous knowledge of vagrant
+to follow the experiment.
+All the config files used in this experiment can be found [here](https://github.com/marpo60/ssh_forward_agent_issue)
+
+Lets start building a topology from scratch.
+
+The topology will include 3 virtual servers and our own machine:
+
+- host
+- web (172.17.8.101)
+- worker (172.17.8.102)
+- secret (172.17.8.110)
+
+Users:
+
+ - me
+ - Alice (Good)
+ - Eve (Bad)
+
+ Facts:
+
+ * The host machine is our machine where we will create the network
+ topology using virtual machines
+ * The web server is a virtual machine that will emulate a server
+ public accesible.
+ * The worker server is a virtual machine that will emulate a
+ server accesible only from the web server.
+ * The secret server is a virtual machine that will emulate a
+ server accessible only by Alice from the internet.
+
+ Lets start the servers with `vagrant up`.
+
+ `host$ vagrant up` (This can take a while...)
+
+ First lets add all the keys to the ssh agent of the host
+ machine
+
+ {% codeblock lang:sh %}
+   me@host$ ssh-add alice_id_rsa
+   me@host$ ssh-add eve_id_rsa
+{% endcodeblock %}
+
+ Now we'll open a new terminal and emulate that Alice is working
+ on the worker server
```

Comment:
I don't understand this exactly. As a user am I supposed to do something on this step?.